File Upload Docs (#866)
ParsePlatform/Docs: _includes/parse-server/file-adapters.md

# Configuring File Upload

> *Available only on Parse Server starting 5.0.0*

Reviewer: ```suggestion
*Available on Parse Server >=5.0.0*
```

# Configuring File Upload

> Starting at Parse Server V5.0.0, Parse Server restricts file upload to authenticated users only to improve Parse Server's default security. This behaviour can be modified by specifying `fileUpload` options to your Parse Server configuration.

Reviewer: Instead of "starting at Parse Server V5.0.0," I think we do have a common tag that we put that indicates the applicable version.

# Configuring File Upload

*Available only on Parse Server starting 5.0.0*

Parse Server restricts file upload to authenticated users only to improve Parse Server's default security. This behaviour can be modified by specifying `fileUpload` options to your Parse Server configuration.

Available options are:

> `enableForAnonymousUser`: Whether file upload should be allowed for anonymous users
> `enableForAuthenticatedUser`: Whether file upload should be allowed for authenticated users
> `enableForPublic`: Whether file upload should be allowed for public users

Reviewer: ```suggestion
- `enableForAnonymousUser`: Enable file upload for anonymous users
- `enableForAuthenticatedUser`: Enable file upload for authenticated users
- `enableForPublic`: Enable file upload for the public, i.e. everyone
```

Reviewer: Did we really call it `enableForPublic` instead of `enableForPublicUser`?

Author: That is what we named it!

To use, simply do any of the following:
- Use the environment variable `PARSE_SERVER_ENCRYPTION_KEY`

> - Pass in --encryptionKey in the command line when starting your server

Reviewer: ```suggestion
- Pass the encryption key via parameter `--encryptionKey` in the command line when starting Parse Server
```

> - Initialize ParseServer with `encryptionKey="Your file encryptionKey"`.

Reviewer: ```suggestion
- Initialize Parse Server with `encryptionKey="PATH_TO_ENCRYPTION_KEY_FILE`.
```
